Lib/pathlib.py

```diff
@@ -989,7 +987,8 @@ def iterdir(self):
     def _scandir(self):
         # Emulate os.scandir(), which returns an object that can be used as a
         # context manager. This method is called by walk() and glob().
-        return contextlib.nullcontext(self.iterdir())
+        from contextlib import nullcontext
+        return nullcontext(self.iterdir())
```

Comment:
You could even consider just reimplementing `nullcontext` inside `pathlib` -- it's not too many lines of code...

```py
class _nullcontext:
    """Vendored from contextlib to avoid depending on contextlib"""

    __slots__ = ("enter_result",)

    def __init__(self, enter_result=None):
        self.enter_result = enter_result

    def __enter__(self):
        return self.enter_result

    def __exit__(self, *args):
        pass
```

What you have is also fine, though, since I doubt this method is particularly performance-critical

Reply:
It's actually impossible to hit this code without using private APIs at the moment, so I might leave it for now!